tests/de.rs

```diff
@@ -220,10 +220,24 @@ mod std_tests {
         assert_eq!(value.unwrap(), Value::Float(100000.0));
     }
 
+    #[cfg(tags)]
     #[test]
     fn test_self_describing() {
         let value: error::Result<Value> =
```

Comment:
Lets change this instead to `Result<String>`, so only a string is expected an we don't need a different test depending on whether tags are enabled.

Reply:
I don't get it. If I would stringify the value using Debug, you would still see the Tag(...) when tags are  enabled and not see the tag when they are disabled. You mean json stringify?

Reply:
I don't want this test for the self-describe tag to vary depending on whether tags are enabled.

```rust
let value: error::Result<String> = de::from_slice(&[0xd9, 0xd9, 0xf7, 0x66, 0x66, 0x6f, 0x6f, 0x62, 0x61, 0x72]);
assert_eq!(value.unwrap(), "foobar".to_owned());
```

Reply:
@rklaehn This is not what I wanted. Doesn't the code work I provided in the previous comment?

Reply:
Unfortunately not. If you compile with tags enabled, a tagged string no longer serialises as a string. The tag causes the `visit_newtype_struct` to be called, which will throw an error for the string deserializer.

See my comment at the end of the discussion thread.